test/unit/io/sequence_file/sequence_file_format_genbank_test.cpp

```diff
@@ -0,0 +1,419 @@
+#include <seqan3/test/pretty_printing.hpp>
+
+using namespace seqan3;
+
+// ----------------------------------------------------------------------------
+// general
+// ----------------------------------------------------------------------------
+
+TEST(general, concepts)
+{
+    EXPECT_TRUE((SequenceFileInputFormat<sequence_file_format_genbank>));
+    EXPECT_TRUE((SequenceFileOutputFormat<sequence_file_format_genbank>));
+}
+
+// ----------------------------------------------------------------------------
+// reading
+// ----------------------------------------------------------------------------
+
+struct read : public ::testing::Test
+{
+    std::vector<std::string> expected_ids
+    {
+        { "ID1" },
+        { "ID2" },
+        { "ID3" },
+    };
+
+    std::vector<dna5_vector> expected_seqs
+    {
+        { "ACGTTTTTTTTTTTTTTT"_dna5 },
+        { "ACGTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT"_dna5 },
+        { "ACGTTTA"_dna5 },
+    };
+
+	std::string input
+    {
+R"(LOCUS ID1 stuff
+DEFINITION  Homo sapiens mRNA for prepro cortistatin like peptide, complete
+            cds.
+ACCESSION   ID1
+ACCESSION   U49845
+VERSION     U49845.1  GI:1293613
+KEYWORDS    .
+SOURCE      Saccharomyces cerevisiae (baker's yeast)
+  ORGANISM  Saccharomyces cerevisiae
+            Eukaryota; Fungi; Ascomycota; Saccharomycotina; Saccharomycetes;
+            Saccharomycetales; Saccharomycetaceae; Saccharomyces.
+REFERENCE   1  (bases 1 to 5028)
+FEATURES             Location/Qualifiers
+     source          1..5028
+ORIGIN
+        1 ACGTTTTTTT TTTTTTTT
+//
+LOCUS ID2
+DEFINITION  Homo sapiens mRNA for prepro cortistatin like peptide, complete
+            cds.
+ACCESSION   ID2
+ORIGIN
+        1  ACGTTTTTTT TTTTTTTTTT TTTTTTTTTT TTTTTTTTTT TTTTTTTTTT TTTTTTTTTT
+       61 TTTTTTTTTT TTTTTTTTTT TT
+//
+LOCUS ID3
+DEFINITION  Homo sapiens mRNA for prepro cortistatin like peptide, complete
+            cds.
+ACCESSION   ID3
+ORIGIN
+        1 ACGTTTA
+//
+)"
+    };
+
+    sequence_file_format_genbank format;
+
+    sequence_file_input_options<dna5, false> options;
+
+    std::string id;
+    dna5_vector seq;
+
+    void do_read_test(std::string const & input)
+    {
+        std::stringstream istream{input};
+
+        for (unsigned i = 0; i < 3; ++i)
+        {
+            id.clear();
+            seq.clear();
+
+            EXPECT_NO_THROW(( format.read(istream, options, seq, id, std::ignore) ));
+            EXPECT_EQ(id, expected_ids[i]);
+            EXPECT_EQ(seq, expected_seqs[i]);
+            EXPECT_TRUE((ranges::equal(seq, expected_seqs[i])));
+            EXPECT_TRUE((ranges::equal(id, expected_ids[i])));
+        }
+    }
+};
+
+TEST_F(read, standard)
+{
+    do_read_test(input);
+}
+
+TEST_F(read, complete_header)
+{
+    options.embl_genbank_complete_header = true;
+    expected_ids[0] = std::string{"LOCUS ID1 stuff\nDEFINITION  Homo sapiens mRNA for prepro cortistatin like peptide"
```

Comment:
maybe you also want to use a raw string literal `R"` here so one can more easily read the header :)